File: CS/WinSample.Module/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp;

namespace WinSample.Module {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            Person john = CreateObject<Person>(new string[] { "FirstName" }, new object[] { "John" });
            Person sam = CreateObject<Person>(new string[] { "FirstName" }, new object[] { "Sam" });

            AnalysedObject analysedObject = CreateObject<AnalysedObject>(new string[] { "Employee", "Year" }, new object[] { john, 2009 });
            if (ObjectSpace.IsNewObject(analysedObject)) {
                analysedObject.BooleanProperty1 = true;
            }
            analysedObject = CreateObject<AnalysedObject>(new string[] { "Employee", "Year" }, new object[] { john, 2010 });
            if (ObjectSpace.IsNewObject(analysedObject)) {
                analysedObject.BooleanProperty2 = true;
            }
            analysedObject = CreateObject<AnalysedObject>(new string[] { "Employee", "Year" }, new object[] { john, 2011 });
            if (ObjectSpace.IsNewObject(analysedObject)) {
                analysedObject.BooleanProperty2 = true;
                analysedObject.BooleanProperty3 = true;
            }
            analysedObject = CreateObject<AnalysedObject>(new string[] { "Employee", "Year" }, new object[] { sam, 2009 });
            if (ObjectSpace.IsNewObject(analysedObject)) {
                analysedObject.BooleanProperty3 = true;
            }
            analysedObject = CreateObject<AnalysedObject>(new string[] { "Employee", "Year" }, new object[] { sam, 2010 });
            if (ObjectSpace.IsNewObject(analysedObject)) {
                analysedObject.BooleanProperty1 = true;
                analysedObject.BooleanProperty2 = true;
            }
            analysedObject = CreateObject<AnalysedObject>(new string[] { "Employee", "Year" }, new object[] { sam, 2011 });
            if (ObjectSpace.IsNewObject(analysedObject)) {
                analysedObject.BooleanProperty1 = true;
            }
        }
        private T CreateObject<T>(string[] propertyNames, object[] propertyValues) where T : XPBaseObject {
            if (propertyValues.Length != propertyNames.Length) {
                throw new ArgumentException();
            }
            GroupOperator criteria = new GroupOperator();
            for (int i = 0; i < propertyNames.Length; i++) {
                criteria.Operands.Add(new BinaryOperator(propertyNames[i], propertyValues[i]));
            }
            T obj = ObjectSpace.FindObject<T>(criteria);
            if (obj == null) {
                obj = ObjectSpace.CreateObject<T>();
                for (int i = 0; i < propertyNames.Length; i++) {
                    obj.ClassInfo.FindMember(propertyNames[i]).SetValue(obj, propertyValues[i]);
                }
            }
            return obj;
        }
    }
}

File: VB/WinSample.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp

Namespace WinSample.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			Dim john As Person = CreateObject(Of Person)(New String() { "FirstName" }, New Object() { "John" })
			Dim sam As Person = CreateObject(Of Person)(New String() { "FirstName" }, New Object() { "Sam" })

			Dim analysedObject As AnalysedObject = CreateObject(Of AnalysedObject)(New String() { "Employee", "Year" }, New Object() { john, 2009 })
			If ObjectSpace.IsNewObject(analysedObject) Then
				analysedObject.BooleanProperty1 = True
			End If
			analysedObject = CreateObject(Of AnalysedObject)(New String() { "Employee", "Year" }, New Object() { john, 2010 })
			If ObjectSpace.IsNewObject(analysedObject) Then
				analysedObject.BooleanProperty2 = True
			End If
			analysedObject = CreateObject(Of AnalysedObject)(New String() { "Employee", "Year" }, New Object() { john, 2011 })
			If ObjectSpace.IsNewObject(analysedObject) Then
				analysedObject.BooleanProperty2 = True
				analysedObject.BooleanProperty3 = True
			End If
			analysedObject = CreateObject(Of AnalysedObject)(New String() { "Employee", "Year" }, New Object() { sam, 2009 })
			If ObjectSpace.IsNewObject(analysedObject) Then
				analysedObject.BooleanProperty3 = True
			End If
			analysedObject = CreateObject(Of AnalysedObject)(New String() { "Employee", "Year" }, New Object() { sam, 2010 })
			If ObjectSpace.IsNewObject(analysedObject) Then
				analysedObject.BooleanProperty1 = True
				analysedObject.BooleanProperty2 = True
			End If
			analysedObject = CreateObject(Of AnalysedObject)(New String() { "Employee", "Year" }, New Object() { sam, 2011 })
			If ObjectSpace.IsNewObject(analysedObject) Then
				analysedObject.BooleanProperty1 = True
			End If
		End Sub
		Private Function CreateObject(Of T As XPBaseObject)(ByVal propertyNames() As String, ByVal propertyValues() As Object) As T
			If propertyValues.Length <> propertyNames.Length Then
				Throw New ArgumentException()
			End If
			Dim criteria As New GroupOperator()
			For i As Integer = 0 To propertyNames.Length - 1
				criteria.Operands.Add(New BinaryOperator(propertyNames(i), propertyValues(i)))
			Next i
			Dim obj As T = ObjectSpace.FindObject(Of T)(criteria)
			If obj Is Nothing Then
				obj = ObjectSpace.CreateObject(Of T)()
				For i As Integer = 0 To propertyNames.Length - 1
					obj.ClassInfo.FindMember(propertyNames(i)).SetValue(obj, propertyValues(i))
				Next i
			End If
			Return obj
		End Function
	End Class
End Namespace
